be/src/storage/chunk_helper.cpp

```diff
@@ -148,8 +148,15 @@ starrocks::vectorized::Schema ChunkHelper::convert_schema_to_format_v2(const sta
 
 starrocks::vectorized::Schema ChunkHelper::get_short_key_schema_with_format_v2(const starrocks::TabletSchema& schema) {
     std::vector<ColumnId> short_key_cids;
-    for (ColumnId cid = 0; cid < schema.num_short_key_columns(); ++cid) {
-        short_key_cids.push_back(cid);
+    const auto& sort_key_idxes = schema.sort_key_idxes();
+    if (!sort_key_idxes.empty()) {
+        for (auto i = 0; i < schema.num_short_key_columns() && i < sort_key_idxes.size(); ++i) {
```

Comment:
why need `i < schema.num_short_key_columns() && i < sort_key_idxes.size()`?

Reply:
it is unnecessary in normal process, but UT can set wrong mock value, in case of crash in UT, i added these.
it can be removed as well because code should not adapt to wrong UT.
changed.